packages/typespec-ts/src/transform/transformResponses.ts

```diff
@@ -104,15 +123,73 @@ export function transformToResponseTypes(
     if (lroLogicalResponse) {
       rlcOperationUnit.responses.push(lroLogicalResponse);
     }
+    rlcOperationUnit.isDefaultSupersetOfOthers =
+      transformIsDefaultSupersetOfOthers(
+        dpgContext,
+        defaultSchemas,
+        nonDefaultSchemas,
+        `${rlcOperationUnit.operationGroup}-${rlcOperationUnit.operationName}`
+      );
     rlcResponses.push(rlcOperationUnit);
   }
   return rlcResponses;
 }
 
+function transformIsDefaultSupersetOfOthers(
+  dpgContext: SdkContext,
+  defaultSchemas: Schema[],
+  nonDefaultSchemas: Schema[],
+  operationName: string
+): boolean {
+  if (defaultSchemas.length !== 1 || nonDefaultSchemas.length === 0) {
+    return false;
+  }
+  const defaultSchema = defaultSchemas[0];
+  if (
+    defaultSchema?.type !== "object" ||
+    !(defaultSchema as ObjectSchema).properties
+  ) {
+    reportDiagnostic(dpgContext.program, {
```

Comment:
Only consider the model subset cases, for default body with other types just report warning.